src/components/common/quantity/Quantity.tsx

```diff
@@ -82,6 +89,12 @@ const Quantity = ({
   };
 
   const parsedValue = quantityUnit?.value || null;
+  const parsedLabel = quantityUnit?.label || null;
+  const [selected, setSelected] = useState<UCUM>({
+    label: parsedLabel,
+    value: parsedValue,
+  });
+  const [value, setValue] = useState<number>(quantityValue);
```

Comment:
I don't think we need a local state to handle the values changed in input, The parent component should maintain the state.